src/storage/PatchStore.ts

```diff
@@ -0,0 +1,46 @@
+import { Conditions } from './Conditions';
+import { Patch } from '../ldp/http/Patch';
+import { PatchHandler } from './patch/PatchHandler';
+import { Representation } from '../ldp/representation/Representation';
+import { RepresentationPreferences } from '../ldp/representation/RepresentationPreferences';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { ResourceStore } from './ResourceStore';
+
+/**
+ * {@link ResourceStore} using decorator pattern for the `modifyResource` function.
+ * If the original store supports the {@link Patch}, behaviour will be identical,
+ * otherwise one of the {@link PatchHandler}s supporting the given Patch will be called instead.
+ */
+export class PatchStore implements ResourceStore {
```

Comment:
Maybe `PatchingStore` or `PatcheableStore` or something (to avoid the impression that it stores patches)?